packages/react/src/components/header/components/headerLink/HeaderLink.tsx

```diff
@@ -159,28 +160,26 @@ export const HeaderLink = <T extends React.ElementType = 'a'>({
   const containerRef = useRef<HTMLSpanElement>(null);
   const isSubNavLink = openSubNavIndex !== undefined && setOpenSubNavIndex !== undefined;
 
-  const handleDynamicMenuPosition = (val: boolean) => {
-    // Null position when false so if user resizes browser, the calculation is done again
-    if (!val) setDynamicPosition(null);
-    // In case of SSR just set the menu to the right
-    else if (val && window === undefined) setDynamicPosition(DropdownMenuPosition.Right);
-    else {
-      // eslint-disable-next-line no-lonely-if
-      if (containerRef.current) {
-        // Calculate which side has more space for the menu
-        const { x: leftPosition, width } = containerRef.current.getBoundingClientRect();
-        const rightPosition = leftPosition + width;
-        const position =
-          leftPosition > window.innerWidth - rightPosition ? DropdownMenuPosition.Left : DropdownMenuPosition.Right;
-        setDynamicPosition(position);
-      }
+  const getDropdownPosition = () => {
+    let position;
```

Comment:
This `let` could be discarded if this function returns the value immediately when resolved.

```
const getDropdownPosition = () => {
    let position;
    // On SSR just set the menu on the right
    if (window === undefined) {
      return DropdownMenuPosition.Right;
    }
    if (containerRef.current != null) {
      const { x: leftPosition, width } = containerRef.current.getBoundingClientRect();
      const rightPosition = leftPosition + width;
      return leftPosition > window.innerWidth - rightPosition ? DropdownMenuPosition.Left : DropdownMenuPosition.Right;
    }
    return position;
  };
```

Btw, it returns undefined, if window is not undefined and containerRef.current is null ? The function or `position`  are not typed, so can't tell is that correct. There should be a ts error setting `undefined` to the `setDynamicPosition`... can't say why it accepts it.

Reply:
I think it's safe to have a default `return DropdownMenuPosition.Right;` at the end.